sbtc/src/leb128.rs

```diff
@@ -26,13 +26,35 @@
 
 use std::io::Cursor;
 
-// LEB128 u64 can use at most 10 bytes
+/// Maximum number of bytes required to encode a u64 in LEB128 format.
+/// For u64 (64 bits), we need at most 10 bytes because each byte provides 7 bits,
+/// with 9 bytes covering 63 bits and the 10th byte providing the final bit.
 const MAX_BYTES: usize = 10;
+
+/// Number of value bits stored in each LEB128 byte.
+/// LEB128 encoding uses 7 bits per byte for actual data, with the 8th bit
+/// serving as a continuation flag.
 const BITS_PER_BYTE: u32 = 7;
+
+/// Maximum number of bits in a u64 value.
+/// Used for bounds checking to prevent overflow during encoding/decoding.
 const MAX_BITS: usize = 64;
+
+/// Bit mask to extract the lower 7 bits (value data) from a LEB128 byte.
+/// Each byte uses bits 0-6 for data and bit 7 for continuation.
 const LOWER_BITS_MASK: u8 = 0x7F;
+
+/// Flag bit indicating that more bytes follow in the LEB128 sequence.
+/// When this bit is set in a byte, it means the value continues in the next byte.
+/// When clear, it indicates the final byte of the encoded sequence.
 const CONTINUATION_FLAG: u8 = 0x80;
 
+/// Represents the threshold where a value needs more than one byte in
+/// Values from 0-127 (0x7F) fit in a single byte, while values ≥128 (0x80)
```

Comment:
qq: what's the "policy" on non ascii chars (`≥`) in the codebase? We do have emojis for logs and ascii art diagrams around, but other than that (and a random `¡`) we don't usually use them.

Reply:
Don't think there's any explicit guidelines/policy; I have `≤≥` on my keyboard so I used them 😅

Reply:
Resolving this, if anyone has an issue with ≤≥ they're welcome to bring it up